datafusion-cli/src/object_storage/instrumented.rs

```diff
@@ -186,6 +209,10 @@ impl ObjectStore for InstrumentedObjectStore {
     }
 
     fn list(&self, prefix: Option<&Path>) -> BoxStream<'static, Result<ObjectMeta>> {
+        if self.enabled() {
+            return self.instrumented_list(prefix);
+        }
+
```

Comment:
I think you may need to instrument the other versions of list below (`list_with_delimiter`, etc) as well

Reply:
Yes, `list_with_delimiter`  is "coming soon to a PR near you" just to help keep the size of the PRs easily reviewable.